archive/tag.go

```diff
@@ -0,0 +1,309 @@
+// Copyright 2024 The NATS Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package archive
+
+import (
+	"fmt"
+)
+
+type TagLabel string
+
+type Tag struct {
+	Name  TagLabel
+	Value string
+}
+
+const (
+	serverTagLabel      TagLabel = "server"
+	clusterTagLabel     TagLabel = "cluster"
+	accountTagLabel     TagLabel = "account"
+	streamTagLabel      TagLabel = "stream"
+	typeTagLabel        TagLabel = "artifact_type"
+	profileNameTagLabel TagLabel = "profile_name"
+)
+
+const (
+	// Server artifacts
+	healtzArtifactType   = "health"
+	varzArtifactType     = "variables"
+	connzArtifactType    = "connections"
+	routezArtifactType   = "routes"
+	gatewayzArtifactType = "gateways"
+	leafzArtifactType    = "leafs"
+	subzArtifactType     = "subs"
+	jszArtifactType      = "jetstream_info"
+	accountzArtifactType = "accounts"
+	// Account artifacts
+	accountConnectionsArtifactType = "account_connections"
+	accountLeafsArtifactType       = "account_leafs"
+	accountSubsArtifactType        = "account_subs"
+	accountJetStreamArtifactType   = "account_jetstream_info"
+	accountInfoArtifactType        = "account_info"
+	streamDetailsArtifactType      = "stream_info"
+	// Other artifacts
+	manifestArtifactType = "manifest"
+	profileArtifactType  = "profile"
+)
+
+const (
+	ManifestFileName = "capture/manifest.json"
+	NoCluster        = "unclustered"
+)
+
+const rootPrefix = "capture/"
+const separator = "__"
+const captureLogName = rootPrefix + "capture.log"
```

Comment:
I went with this because i wasn't sure of portability. 
Filepath separator is `\` on windows, and ZIP always want `/`.
However after a little research I see `path` package always uses `/`, so going to go with `path.Join`